addOns/openapi/src/main/java/org/zaproxy/zap/extension/openapi/OpenApiSpider.java

```diff
@@ -65,10 +65,15 @@ public boolean parseResource(HttpMessage message, Source source, int depth) {
     public boolean canParseResource(HttpMessage message, String path, boolean wasAlreadyConsumed) {
         try {
             String contentType = message.getResponseHeader().getHeader(HttpHeader.CONTENT_TYPE);
-            if (contentType.toLowerCase().startsWith("text")
-                    && message.getResponseBody().toString().toLowerCase().startsWith("swagger:")) {
+            if (contentType.toLowerCase().startsWith("application/vnd.oai.openapi")) {
                 return true;
-            } else if (contentType.toLowerCase().startsWith("application")) {
+            } else if ((contentType.toLowerCase().contains("json")
+                            || contentType.toLowerCase().contains("yaml"))
+                    && (message.getResponseBody().toString().toLowerCase().startsWith("swagger:")
```

Comment:
Calling message.getResponseBody().toString().toLowerCase() twice seems very inefficient. Maybe do it just once, and could extract the first 8 chrs rather than doing toLowerCase() on a potentially very large response?